src/plugins/unified_field_list/common/utils/field_examples_calculator.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// Adapted from src/plugins/discover/public/application/main/components/sidebar/lib/field_calculator.js
+
+import { map, sortBy, defaults, isObject, pick } from 'lodash';
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { DataView, DataViewField } from '@kbn/data-plugin/common';
+import { flattenHit } from '@kbn/data-plugin/common';
+
+type FieldHitValue = any;
+
+interface FieldValueCountsParams {
+  hits: estypes.SearchHit[];
+  dataView: DataView;
+  field: DataViewField;
+  count?: number;
+}
+
+export const canFetchFieldExamples = (field: DataViewField): boolean => {
+  return !(
+    field.type === 'geo_point' ||
+    field.type === 'geo_shape' ||
+    field.type === 'attachment' ||
+    field.type === 'unknown'
+  );
+};
+
+export function getFieldExampleBuckets(params: FieldValueCountsParams) {
+  params = defaults(params, {
+    count: 5,
+  });
+
+  if (!canFetchFieldExamples(params.field)) {
+    throw new Error('Analysis is not available this field type');
+  }
+
+  const records = getFieldValues(params.hits, params.field, params.dataView);
+  const { groups, sampledValues } = groupValues(records);
+  const buckets = sortBy(groups, ['count', 'order'])
+    .reverse()
+    .slice(0, params.count)
+    .map((bucket) => pick(bucket, ['key', 'count']));
+
+  if (!sampledValues) {
+    throw new Error('No data for this field in the first found records');
+  }
```

Comment:
For both Errors we could add the name of the field. Just for the case the error is not seen immediately, one might wonder, what field triggered it ... however definitely not a necessary change , file under loud thinking